### PR TITLE
Added wow_buy_token and wow_sell_token actions

### PR DESCRIPTION
**Pull request by rohan-agarwal-coinbase**
### What changed? Why?
Added `wow_buy_token` and `wow_sell_token`.py

### Tests
- Unit tests pass
- E2E testing passes where I instruct the chatbot to buy tokens and sell tokens

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: cdp-agentkit-core/cdp_agentkit_core/actions/wow/buy_token.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+class WowBuyTokenInput(BaseModel):
+    """Input argument schema for buy token action."""
+
+    contract_address: str = Field(
+        ...,
+        description="The WOW token contract address, such as `0x036CbD53842c5426634e7929541eC2318f3dCF7e`",
+    )
+
+    amount_eth: str = Field(
```
Comment (alex-stone) [inline cdp-agentkit-core/cdp_agentkit_core/actions/wow/buy_token.py:26]:
```suggestion
    amount_eth: str = Field(
```

Should this be `amount_wei`, curious if `amount_eth` would be part of the reason it is confused?

Comment (rohan-agarwal-coinbase) [reply cdp-agentkit-core/cdp_agentkit_core/actions/wow/buy_token.py:26]: yeah it's no longer confused now, but maybe that helps further? will make the change!

File: cdp-agentkit-core/cdp_agentkit_core/actions/wow/buy_token.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        contract_address=contract_address,
+        method="buy",
+        abi=WOW_ABI,
+        args={
+            "recipient": wallet.default_address.address_id,
+            "refundRecipient": wallet.default_address.address_id,
+            "orderReferrer": "0x0000000000000000000000000000000000000000",
+            "expectedMarketType": has_graduated and "1" or "0",
+            "minOrderSize": min_tokens,
+            "sqrtPriceLimitX96": "0",  # TODO
```
Comment (John-peterson-coinbase) [inline cdp-agentkit-core/cdp_agentkit_core/actions/wow/buy_token.py:60]: Can we add a more descriptive TODO here.

File: cdp-agentkit-core/cdp_agentkit_core/actions/wow/quotes.py
Comment (John-peterson-coinbase) [inline cdp-agentkit-core/cdp_agentkit_core/actions/wow/quotes.py:1]: Can we make it clear in the directory structure that these are utils?

File: cdp-agentkit-core/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@
 ### Added
 
 - Enhanced `wow_create_token` action error handling.
+- Added `wow_buy_token` and `wow_sell_token`.
```
[the remark this answers is not in the document]
Comment (rohan-agarwal-coinbase) [reply cdp-agentkit-core/CHANGELOG.md:10]: oops issue with the merge, good catch fixed

File: cdp-agentkit-core/cdp_agentkit_core/actions/wow/buy_token.py (hunk 2 above)
Comment (rohan-agarwal-coinbase) [reply cdp-agentkit-core/cdp_agentkit_core/actions/wow/buy_token.py:60]: removed the TODO - double checked with zora in that thread that 0 is fine

**Comment by John-peterson-coinbase** [comment]
Looks good oerall. We need to add the new actions to the documentation in `cdp-langchain` 

https://github.com/coinbase/cdp-agentkit/blob/90a8140c31b3c102571a1298fa261460518084f9/cdp-langchain/cdp_langchain/agent_toolkits/cdp_toolkit.py#L63
https://github.com/coinbase/cdp-agentkit/blob/90a8140c31b3c102571a1298fa261460518084f9/cdp-langchain/README.md?plain=1#L62
https://github.com/coinbase/cdp-agentkit/blob/90a8140c31b3c102571a1298fa261460518084f9/README.md?plain=1#L19

cc: @rohan-agarwal-coinbase

File: cdp-agentkit-core/cdp_agentkit_core/actions/wow/sell_token.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    Returns:
+        str: A message confirming the sale with the transaction hash
+
+    """
+    eth_quote = get_sell_quote(wallet.network_id, contract_address, amount_tokens_in_wei)
+    has_graduated = get_has_graduated(wallet.network_id, contract_address)
+
+    # Multiply by 98/100 and floor to get 98% of quote as minimum (slippage protection)
+    min_eth = str(int((eth_quote * 98) // 100))
```
[the remark this answers is not in the document]
Comment (rohan-agarwal-coinbase) [reply cdp-agentkit-core/cdp_agentkit_core/actions/wow/sell_token.py:48]: will ask the zora team if there was a good reason for this

[the remark this answers is not in the document]
Comment (rohan-agarwal-coinbase) [reply cdp-agentkit-core/cdp_agentkit_core/actions/wow/sell_token.py:48]: will ask the zora team if there was a good reason for this

File: cdp-agentkit-core/cdp_agentkit_core/actions/wow/quotes.py
Comment (rohan-agarwal-coinbase) [reply cdp-agentkit-core/cdp_agentkit_core/actions/wow/quotes.py:1]: renamed the file `utils.py`, feels cleaner than a separate folder

File: cdp-agentkit-core/cdp_agentkit_core/actions/wow/sell_token.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (rohan-agarwal-coinbase) [reply cdp-agentkit-core/cdp_agentkit_core/actions/wow/sell_token.py:48]: i'll land this as is for now, but will follow up and standardize on 98 or 99 depending on what they say

**Review by alex-stone — APPROVED** [review]

**Review by alex-stone — APPROVED** [review]

**Review by John-peterson-coinbase — APPROVED** [review]

**Pull request merged**